### ISSUE TITLE
Too many tokens

### ISSUE THREAD

**Issue by integral-llc** [codelion/optillm#4]
I'm using the default sample client you've shown, and getting this error with `moa` and `cot_reflection` approaches.
```
2024-09-12 15:18:25,087 - INFO - 127.0.0.1 - - [12/Sep/2024 15:18:25] "POST /v1/chat/completions HTTP/1.1" 500 -
2024-09-12 15:18:26,718 - INFO - Received request to /v1/chat/completions
2024-09-12 15:18:26,719 - INFO - Using approach cot_reflection, with gpt-4o
2024-09-12 15:18:27,008 - INFO - HTTP Request: POST https://api.openai.com/v1/chat/completions "HTTP/1.1 400 Bad Request"
2024-09-12 15:18:27,013 - ERROR - Error processing request: Error code: 400 - {'error': {'message': 'max_tokens is too large: 8192. This model supports at most 4096 completion tokens, whereas you provided 8192.', 'type': 'invalid_request_error', 'param': 'max_tokens', 'code': None}}
202
```

**Comment by codelion** [comment]
You are right the max_token output limit for gpt-4o is 4096 I will fix it shortly.

**Comment by codelion** [comment]
fixed in [15b7fe4e716560a033827bb4b478dacf08952b19](https://github.com/codelion/optillm/commit/15b7fe4e716560a033827bb4b478dacf08952b19)
